pkg/generators/system/generator.go

```diff
@@ -466,14 +489,49 @@ func (gen *SphinxGenerator) StatefulSet() basereconciler_resources.StatefulSetTe
 		RolloutTriggers: []basereconciler_resources.RolloutTrigger{
 			{Name: "system-database", SecretName: pointer.String("system-database")},
 		},
-		IsEnabled: true,
+		IsEnabled: gen.Enabled,
 	}
 }
 
 func (gen *SphinxGenerator) Service() basereconciler_resources.ServiceTemplate {
 	return basereconciler_resources.ServiceTemplate{
 		Template:  gen.service(),
-		IsEnabled: true,
+		IsEnabled: gen.Enabled,
+	}
+}
+
+// SearchdGenerator has methods to generate resources for system-Searchd
+type SearchdGenerator struct {
+	generators.BaseOptionsV2
+	Spec                 saasv1alpha1.SystemSearchdSpec
+	Image                saasv1alpha1.ImageSpec
+	DatabasePort         int32
+	DatabasePath         string
+	DatabaseStorageSize  resource.Quantity
+	DatabaseStorageClass *string
+	Enabled              bool
+}
+
+func (gen *SearchdGenerator) StatefulSetWithTraffic() []basereconciler.Resource {
+	return []basereconciler.Resource{
+		gen.StatefulSet(), gen.Service(),
+	}
+}
+
+func (gen *SearchdGenerator) StatefulSet() basereconciler_resources.StatefulSetTemplate {
+	return basereconciler_resources.StatefulSetTemplate{
+		Template: gen.statefulset(),
+		RolloutTriggers: []basereconciler_resources.RolloutTrigger{
+			{Name: "system-database", SecretName: pointer.String("system-database")},
```

Comment:
What does this mean? That the rollout is triggered when the `system-database` secret changes?

I believe searchd component does not depend on this, because it will not connect to the DB (the config will be pre-generated).

Reply:
No. We only use real-time indices so search never connects to MySQL. You can check changes in on-prem operator:

- no variables need to be passed to pod
- volume should be mounted in a different path
- no command and command line arguments need to be passed to the pod

It is basically a very standalone component.

Previously probably it made sense for us, because we triggered indexing on the sphinx pod itself, which was performed by rails reaching out to MySQL and then writing data to sphinx.

Now though, the searchd image is  bare RHEL7 image without ruby, rails and many other things needed. So indexing cannot happen from searchd pod. Indexing needs to happen from a functional rails pod (system-console-0 including).

HTH

PS great catch @mayorova

Reply:
I wouldn't honestly try to understand all changes in code here. It would be much easier to understand looking at the end result of the deployment. The serachd stateful set and resulting pod.